arroyo/processing/strategies/decoder/base.py

```diff
@@ -39,14 +39,17 @@ def poll(self) -> None:
         self.__next_step.poll()
 
     def submit(self, message: Message[KafkaPayload]) -> None:
-        decoded_value = self.__codec.decode(
-            message.payload.value, validate=self.__validate
-        )
-
-        decoded = DecodedKafkaMessage(
-            message.payload.key, decoded_value, message.payload.headers
-        )
-        self.__next_step.submit(Message(message.value.replace(decoded)))
+        if isinstance(message.payload, FilteredPayload):
+            self.__next_step.submit(cast(Message[DecodedKafkaMessage[T]], message))
```

Comment:
this cast is there to convert `message` from `Message[TPayload]` to `Message[DecodedKafkaMessage[T]]`. This is fine because the payload is actually `FilteredPayload`.

This cast used to be hidden within a method on `Message`.